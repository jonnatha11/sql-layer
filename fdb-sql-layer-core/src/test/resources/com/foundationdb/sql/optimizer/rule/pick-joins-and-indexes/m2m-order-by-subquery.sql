-- Make sure that this is ordered correctly
-- The limit 1000000 is to make sure that subquery stays a subquery even if we improve the flattener
-- I tried removing the subquery and it caused this to work correctly, see m2m-order-by
SELECT b.id AS b_id, b.data AS b_data, anon_1.a_id AS anon_1_a_id, m2m_1.id AS m2m_1_id
    FROM (SELECT a.id AS a_id FROM a LIMIT 10000000) AS anon_1
    JOIN m2m AS m2m_1 ON anon_1.a_id = m2m_1.aid
    JOIN b ON b.id = m2m_1.bid
    ORDER BY anon_1.a_id, m2m_1.id
