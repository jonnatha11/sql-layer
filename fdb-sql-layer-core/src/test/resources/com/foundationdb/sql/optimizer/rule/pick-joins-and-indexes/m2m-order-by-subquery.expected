SelectQuery@76b3eddb
  ResultSet@1ec501bf[b_id, b_data, anon_1_a_id, m2m_1_id]
    Project@17f5b261[b.id, b.data, anon_1[0], m2m_1.id]
      Sort@3920a91[anon_1[0], m2m_1.id]
        Select@68f3b76b[]
          JoinNode@1e36505c(INNER/NESTED_LOOPS[m2m_1.bid == b.id])
            TableGroupJoinTree@4939c302(TableGroup@2bfd5f0e(b), b - GroupScan@20062487(Group(test.b), rows = 1, cost = 3.00000))
              TableSource@6c0e42c0(b - TableGroup@2bfd5f0e(b))
            JoinNode@7fd5d782(INNER/NESTED_LOOPS[m2m_1.aid == anon_1[0]])
              TableGroupJoinTree@3ebe0a6a(TableGroup@200ece1b(m2m), m2m_1 - GroupScan@37b60886(Group(test.m2m), rows = 1, cost = 4.00000))
                TableSource@7fd0e77e(m2m_1 - TableFKJoin@32dccf8c(Foreign Key m2m_fkey$1: test.m2m REFERENCES test.b))
              SubquerySource@429cc16d(anon_1)
                Subquery@415efd45
                  ResultSet@4345de8a[a_id]
                    Limit@20155887(LIMIT 10000000)
                      Project@59671785[a.id]
                        Select@7f394a9d[]
                          TableGroupJoinTree@2e9dd554(TableGroup@7c2095b4(a), a - SingleIndexScan@6ca3b95(Index(test.a.PRIMARY[IndexColumn(id)]), covering/NONE, rows = 2, cost = 8.17960))
                            TableSource@a51caac(a - TableGroup@7c2095b4(a))